flutter/lib/src/sentry_flutter.dart

```diff
@@ -110,18 +116,21 @@ mixin SentryFlutter {
     // The ordering here matters, as we'd like to first start the native integration.
     // That allow us to send events to the network and then the Flutter integrations.
     // Flutter Web doesn't need that, only Android and iOS.
-    if (options.platformChecker.hasNativeIntegration) {
-      integrations.add(NativeSdkIntegration(channel));
+    if (options.isNativeIntegrationAvailableAndEnabled &&
+        options.autoInitializeNative) {
+      integrations.add(InitNativeSdkIntegration(channel));
```

Comment:
Instead of adding or not the integration, let it be added, but during the installation of the integration, you check the flag and bail out if disabled.